depot/src/main/java/no/hyper/depot/Depot.java

```diff
@@ -0,0 +1,158 @@
+package no.hyper.depot;
+
+import android.content.Context;
+import android.util.Log;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by espenalmdahl on 18/06/15.
+ */
+public class Depot {
+
+    private static Depot singleton;
+    private Context context;
+    private static final String TAG = "==> DEPOT <==";
+
+    private Depot(Context context) {
+        this.context = context.getApplicationContext();
+    }
+
+    public static Depot with(Context context) {
+        if (singleton == null) {
+            singleton = new Depot(context);
+        }
+        return singleton;
+    }
+
+
+    /**
+     * Store a binary object. Remember to implement the Serializable interface in all
+     * referenced objects
+     *
+     * @param name
+     * @param serializableObject
+     */
+    public void store(String name, Object serializableObject) {
+        if ( serializableObject instanceof Serializable) {
+            try {
+                BufferedOutputStream bos = new BufferedOutputStream(context.openFileOutput(name, Context.MODE_PRIVATE));
+                ObjectOutputStream oos = new ObjectOutputStream(bos);
+                oos.writeObject(serializableObject);
```

Comment:
This operation can be quite time consuming. Because writing to a file involves I/O operation. The file would be **corrupted** if app crashes while it's writing. A safer solution would be. 1) write to a *_tmp file 2) write the bytes to the temp file. 3) rename the *_tmp to the preferred name. So in case of crash during process step 1/2, it's only treated as a "garbage file". For step 3, from the OS perspective, the renaming process is atomic operation.